### ISSUE TITLE
Crash when portal does not lead to anywhere

### ISSUE THREAD

**Issue by timothyqiu** [godotengine/godot#51390]
### Godot version

3.4.beta3

### System information

macOS 11.5.1

### Issue description

Encountered a crash when trying to make an MRP for #51389 :)

There are two portals that does not lead to anywhere because I deleted the outer scene.

<details>

<summary>Here is the backtrace</summary>

```
convert_roomgroup : Building
		AUTOLINK OK from EntranceHall to Hallway
WARNING: Portal AUTOLINK failed for Portal_Outdoor from Yard
     at: _autolink_portals (scene/3d/room_manager.cpp:1020)
WARNING: Portal AUTOLINK failed for Portal_Outdoor2 from Yard
     at: _autolink_portals (scene/3d/room_manager.cpp:1020)
ROOM	EntranceHall
		POUT	Hallway
		POUT	Yard
		PIN 	Yard
		PIN 	Yard
			GEOM	Floor
			GEOM	Floor3
			GEOM	Floor2
			GEOM	Floor4
			GEOM	Floor5
ROOM	Hallway
		PIN 	Room
		PIN 	EntranceHall
			GEOM	CSGPolygon
			GEOM	CSGPolygon2
			GEOM	CSGPolygon3
			GEOM	CSGPolygon4
ROOM	Room
		POUT	Hallway
			GEOM	Floor
			GEOM	Floor6
			GEOM	Floor2
			GEOM	Floor3
			GEOM	Floor4
			GEOM	Floor5
ROOM	Yard
		PIN 	EntranceHall
		POUT	EntranceHall
		POUT	EntranceHall
handle_crash: Program crashed with signal 11
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] 1   libsystem_platform.dylib            0x00007fff203d6d7d _sigtramp + 29
[2] 2   libsystem_c.dylib                   0x00007fff8068a320 usual + 0
[3] StringName::StringName(StringName const&) (in godot.osx.tools.64) (string_name.cpp:156)
[4] Node::get_name() const (in godot.osx.tools.64) (node.cpp:904)
[5] RoomManager::_find_name_before(Node*, String, bool) (in godot.osx.tools.64) (room_manager.cpp:2038)
[6] RoomManager::_third_pass_rooms(LocalVector<Portal*, unsigned int, false> const&) (in godot.osx.tools.64) (room_manager.cpp:854)
[7] RoomManager::rooms_convert() (in godot.osx.tools.64) (room_manager.cpp:630)
[8] MethodBind0::call(Object*, Variant const**, int, Variant::CallError&) (in godot.osx.tools.64) (method_bind.gen.inc:59)
[9] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (in godot.osx.tools.64) (object.cpp:918)
[10] Variant::call_ptr(StringName const&, Variant const**, int, Variant*, Variant::CallError&) (in godot.osx.tools.64) (variant_call.cpp:1173)
[11] GDScriptFunction::call(GDScriptInstance*, Variant const**, int, Variant::CallError&, GDScriptFunction::CallState*) (in godot.osx.tools.64) (gdscript_function.cpp:1034)
[12] GDScriptInstance::_ml_call_reversed(GDScript*, StringName const&, Variant const**, int) (in godot.osx.tools.64) (gdscript.cpp:1181)
[13] GDScriptInstance::call_multilevel_reversed(StringName const&, Variant const**, int) (in godot.osx.tools.64) (gdscript.cpp:1189)
[14] Node::_notification(int) (in godot.osx.tools.64) (node.cpp:149)
[15] Node::_notificationv(int, bool) (in godot.osx.tools.64) (node.h:45)
[16] Spatial::_notificationv(int, bool) (in godot.osx.tools.64) + 63
[17] Object::notification(int, bool) (in godot.osx.tools.64) (object.cpp:927)
[18] Node::_propagate_ready() (in godot.osx.tools.64) (node.cpp:187)
[19] Node::_propagate_ready() (in godot.osx.tools.64) (node.cpp:178)
[20] Node::_set_tree(SceneTree*) (in godot.osx.tools.64) (node.cpp:0)
[21] SceneTree::init() (in godot.osx.tools.64) (scene_tree.cpp:464)
[22] OS_OSX::run() (in godot.osx.tools.64) (os_osx.mm:3233)
[23] main (in godot.osx.tools.64) (godot_main_osx.mm:71)
[24] 24  libdyld.dylib                       0x00007fff203acf3d start + 1
[25] 25  ???                                 0x0000000000000003 0x0 + 3
-- END OF BACKTRACE --
Abort trap: 6
```

</details>

### Steps to reproduce

Run the MRP.

### Minimal reproduction project

[PortalCrash.zip](https://github.com/godotengine/godot/files/6950749/PortalCrash.zip)

**Comment by lawnjelly** [comment]
Good job finding this, it sounds like a dangling reference. I have to unload the portal system when certain objects are deleted, sounds like one that I didn't catch.

**Comment by timothyqiu** [comment]
Just tried your fix and it indeed fixed the crash 🎉

**Comment by akien-mga** [comment]
Fixed by #51394.
